java/core/pom.xml

```diff
@@ -97,6 +97,11 @@
       <artifactId>mockito-core</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>io.airlift</groupId>
+      <artifactId>slice</artifactId>
+      <scope>test</scope>
```

Comment:
Unfortunately, this isn't a test scope, but an actual dependence. I tried to use a zstd file and it failed with a class not found exception.

I opened https://github.com/airlift/slice/issues/119 to discuss the problem with the slice maintainers.

As it stands, we can't add this to the build for Java. The best we can do is add an additional non-default maven profile that will pull in the GPL infected code. *sigh*

Reply:
They replied that jot-core is actually GPL with a classpath exemption. Unfortunately, that is still [category X](https://www.apache.org/legal/resolved.html#category-x) from Apache's point of view. We could either:
* Create a non-default maven profile that enables zstd and pulls in the GPL'ed code.
* Extend the shim layer to detect whether jot-core is available on the classpath and enable the zstd code at runtime. If we go that way, I'd recommend having the aircompressor block jot-core rather than slice and add jot-core as an optional dependency.

Reply:
Ok, looking through the Apache legal discussions, I found the relevant thread about using [GPL+CPE in Apache projects](https://issues.apache.org/jira/browse/LEGAL-336).

Reply:
@omalley So I still have to choose either way you provided above, right?

Reply:
We either need to wait until aircompressor releases with https://github.com/airlift/aircompressor/pull/100 or we need to use one of the approaches above.

Reply:
The summary of aircompressor#100 is that they remove the dependency on slice completely by pulling a hash function over.